ports/posix/iot_logging.c

```diff
@@ -0,0 +1,405 @@
+    #endif
+#else /* if IOT_STATIC_MEMORY_ONLY == 1 */
+    #ifndef IotLogging_Malloc
+        #include <stdlib.h>
+        #define IotLogging_Malloc    malloc
+    #endif
+
+    #ifndef IotLogging_Free
+        #include <stdlib.h>
+        #define IotLogging_Free    free
+    #endif
+#endif /* if IOT_STATIC_MEMORY_ONLY == 1 */
+
+/**
+ * @brief A guess of the maximum length of a timestring.
+ *
+ * There's no way for this logging library to know the length of a timestring
+ * before it's generated. Therefore, the logging library will assume a maximum
+ * length of any timestring it may get. This value should be generous enough
+ * to accommodate the vast majority of timestrings.
+ *
+ * @see @ref platform_clock_function_gettimestring
+ */
+#define MAX_TIMESTRING_LENGTH    ( 64 )
+
+/**
+ * @brief The longest string in #_pLogLevelStrings (below), plus 3 to accommodate
+ * `[]` and a null-terminator.
+ */
+#define MAX_LOG_LEVEL_LENGTH     ( 8 )
+
+/**
+ * @brief How many bytes @ref logging_function_genericprintbuffer should output on
+ * each line.
+ */
+#define BYTES_PER_LINE           ( 16 )
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Lookup table for log levels.
+ *
+ * Converts one of the @ref logging_constants_levels to a string.
+ */
+static const char * const _pLogLevelStrings[ 5 ] =
+{
+    "",      /* IOT_LOG_NONE */
+    "ERROR", /* IOT_LOG_ERROR */
+    "WARN ", /* IOT_LOG_WARN */
+    "INFO ", /* IOT_LOG_INFO */
+    "DEBUG"  /* IOT_LOG_DEBUG */
+};
+
+/*-----------------------------------------------------------*/
+
+#if !defined( IOT_STATIC_MEMORY_ONLY ) || ( IOT_STATIC_MEMORY_ONLY == 0 )
+    static bool _reallocLoggingBuffer( void ** pOldBuffer,
+                                       size_t newSize,
+                                       size_t oldSize )
+    {
+        bool status = false;
+
+        /* Allocate a new, larger buffer. */
+        void * pNewBuffer = IotLogging_Malloc( newSize );
+
+        /* Ensure that memory allocation succeeded. */
+        if( pNewBuffer != NULL )
+        {
+            /* Copy the data from the old buffer to the new buffer. */
+            ( void ) memcpy( pNewBuffer, *pOldBuffer, oldSize );
+
+            /* Free the old buffer and update the pointer. */
+            IotLogging_Free( *pOldBuffer );
+            *pOldBuffer = pNewBuffer;
+
+            status = true;
+        }
+
+        return status;
+    }
+#endif /* if !defined( IOT_STATIC_MEMORY_ONLY ) || ( IOT_STATIC_MEMORY_ONLY == 0 ) */
+
+/*-----------------------------------------------------------*/
+
+void IotLog_Generic( int32_t messageLevel,
+                     const char * const pFormat,
+                     ... )
+{
+    int requiredMessageSize = 0;
+    size_t bufferSize = 0,
+           bufferPosition = 0, timestringLength = 0;
+    char * pLoggingBuffer = NULL;
+    va_list args;
+
+    if( messageLevel == 0 )
```

Comment:
This wont be possible anymore as the code is now compiled out in case of 0 error.

Reply:
Removed